src/robusta/integrations/jira/client.py

```diff
@@ -131,6 +153,133 @@ def _get_default_project_id(self):
+                    }
+                ],
+                "type": "doc",
+                "version": 1,
+            }
+        }
+        logging.debug(f"Adding comment to issue {issue_id} with payload: {payload}")
+        response = self._call_jira_api(url, http_method=HttpMethod.POST, json=payload) or {}
+        logging.debug(response)
+
+    def create_issue(self, issue_data, issue_attachments=None):
+        endpoint = "issue"
+        url = self._get_full_jira_url(endpoint)
+        payload = {
+            "update": {},
+            "fields": {
+                **issue_data,
+                "issuetype": {"id": str(self.default_issue_type_id)},
+                "project": {"id": str(self.default_project_id)},
+            },
+        }
+        logging.debug(f"Create issue with payload: {payload}")
+        response = self._call_jira_api(url, http_method=HttpMethod.POST, json=payload) or {}
+        logging.debug(response)
+        issue_id = response.get("id")
+        if issue_id and issue_attachments:
+            self.add_attachment(issue_id, issue_attachments)
+
+    def update_issue(self, issue_id, issue_data):
+        summary = self._get_nested_property(issue_data, "summary", "")
+        description = self._get_nested_property(issue_data, "description", "")
+        endpoint = f"issue/{issue_id}"
+        url = self._get_full_jira_url(endpoint)
+
+        payload = {
+            "fields": {"summary": summary, "description": description},
+        }
+
+        logging.debug(f"Update issue '{issue_id}' with payload: {payload}")
+        response = self._call_jira_api(url, http_method=HttpMethod.PUT, json=payload) or {}
+        logging.debug(response)
+
+    def manage_issue(self, issue_data, alert_data, issue_attachments=None):
+        existing_issue = self._get_issue_for_labels(issue_data.get("labels", [])) or []
+        status = self._get_nested_property(alert_data, "status", -1)
+        source = self._get_nested_property(alert_data, "source", None)
+        alert_resolved = status == FindingStatus.RESOLVED
+        is_prometheus_alert = source == FindingSource.PROMETHEUS
+
+        if not is_prometheus_alert:
+            # It's not an alert fired from Prometheus, we simply create
+            # a Jira issue without other logic involved
+            self.create_issue(issue_data, issue_attachments)
+        elif existing_issue:
+            issue_done = self._check_issue_done(existing_issue)
+            issue_id = self._get_nested_property(existing_issue, "id", -1)
+            issue_resolution = self._get_nested_property(existing_issue, "fields.resolution.name", "")
+
+            if issue_done:
+                if alert_resolved:
+                    logging.info(
+                        f"Ignoring resolved alert that is already 'done' in Jira for issue with id '{issue_id}'"
+                    )
+                elif self.noReopenResolution and issue_resolution == self.noReopenResolution:
```

Comment:
This is a new piece of logic that I've introduced. I think that it could be useful.
It allows anyone to stop Robusta from re-opening issues on an issue basis.

For example, let's say that I've an issue that I know I won't fix, and that issue is closed with a "Won't Fix" resolution on Jira. Then I could use this resolution with the `noReopenResolution` JiraParam to prevent Robusta from re-opening it.